components/brave_wallet/browser/meld_integration_service.cc

```diff
@@ -0,0 +1,549 @@
+        data:
+          "Meld JSON RPC response bodies."
+        destination: WEBSITE
+      }
+      policy {
+        cookies_allowed: NO
+        setting:
+          "You can enable or disable this feature on chrome://flags."
+        policy_exception_justification:
+          "Not implemented."
+      }
+    )");
+}
+
+base::flat_map<std::string, std::string> MakeMeldApiHeaders() {
+  base::flat_map<std::string, std::string> request_headers;
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  std::string meld_api_key(BUILDFLAG(MELD_API_KEY));
+  if (env->HasVar("MELD_API_KEY")) {
+    env->GetVar("MELD_API_KEY", &meld_api_key);
+  }
+  meld_api_key = base::StrCat({"BASIC", " ", meld_api_key});
+  request_headers["Authorization"] = std::move(meld_api_key);
+  request_headers["accept"] = "application/json";
+  request_headers[brave_wallet::kMeldRpcVersionHeader] =
+      brave_wallet::kMeldRpcVersion;
+
+  return request_headers;
+}
+constexpr char kDefaultMeldStatuses[] = "LIVE,RECENTLY_ADDED";
+
+std::optional<std::string> SanitizeJson(const std::string& json) {
+  auto converted_json = std::string(
+      json::convert_all_numbers_to_string_and_remove_null_values(json, ""));
+  if (converted_json.empty()) {
+    return std::nullopt;
+  }
+
+  return converted_json;
+}
+
+GURL AppendFilterParams(
+    GURL url,
+    brave_wallet::mojom::MeldFilterPtr filter,
+    std::optional<base::flat_map<std::string, std::string>> def_params) {
+  if (def_params) {
+    for (const auto& [key, val] : *def_params) {
+      url = net::AppendQueryParameter(url, key, val);
+    }
+  }
+
+  url = net::AppendQueryParameter(
+      url, "statuses",
+      filter ? filter->statuses.value_or(kDefaultMeldStatuses)
+             : kDefaultMeldStatuses);
+
+  if (!filter) {
+    return url;
+  }
+
+  if (filter->countries) {
+    url = net::AppendQueryParameter(url, "countries", *filter->countries);
+  }
+  if (filter->fiat_currencies) {
+    url = net::AppendQueryParameter(url, "fiatCurrencies",
+                                    *filter->fiat_currencies);
+  }
+  if (filter->crypto_currencies) {
+    url = net::AppendQueryParameter(url, "cryptoCurrencies",
+                                    *filter->crypto_currencies);
+  }
+  if (filter->service_providers) {
+    url = net::AppendQueryParameter(url, "serviceProviders",
+                                    *filter->service_providers);
+  }
+  if (filter->payment_method_types) {
+    url = net::AppendQueryParameter(url, "paymentMethodTypes",
+                                    *filter->payment_method_types);
+  }
+  return url;
+}
+
+bool NeedsToParseResponse(const int& http_error_code) {
+  static const base::NoDestructor<std::unordered_set<int>>
+      kRespCodesAllowedToContinueParsing({400, 401, 403});
+  return kRespCodesAllowedToContinueParsing->contains(http_error_code);
+}
+
+}  // namespace
+
+namespace brave_wallet {
+using std::move;
```

Comment:
I couldn't quickly find guide on this, but chromium does not use `using std::move;`. Let's follow that.